app/src/utils/api.ts

```diff
@@ -0,0 +1,43 @@
+import data from "../../src/data/lanes";
+import { getItemsCountFromUUIDs } from "../../src/utils/utils";
+import { NextResponse, NextRequest } from "next/server";
+import type { LaneDataType } from "../../src/types/Lane";
+export const fetchCache = "force-no-store";
+
+const homePageData = async () => {
+  const randomNumber = Math.floor(Math.random() * 2);
+  const lanes: LaneDataType[] = data.lanes as unknown as LaneDataType[];
+
+  // Get all the UUIDs from the collections
+  const allCollectionUUIDs: string[] = lanes.reduce((acc, lane) => {
+    return acc.concat(lane.collections.map((collection) => collection.uuid));
+  }, [] as string[]);
+  const uuidtoItemCountMap = await getItemsCountFromUUIDs(allCollectionUUIDs);
+
+  // Update the collections for each lane with the number of items
+  const updatedLanes = lanes.map((lane) => {
+    const updatedCollections = lane.collections.map((collection) => {
+      return {
+        ...collection,
+        numItems: uuidtoItemCountMap[collection.uuid] || "0",
+      };
+    });
+    return { ...lane, collections: updatedCollections };
+  });
+
+  const newResponse = { randomNumber, lanesWithNumItems: updatedLanes };
+  return JSON.parse(JSON.stringify(newResponse));
+};
+
+export const getHomePageData = async () => {
+  const data = await homePageData();
+  console.log("data in utils/api.ts is :", data);
+  if (data.lanesWithNumItems[0]) {
+    return data;
+  } else {
+    console.log("There is an error getting the home page data: ", data);
+    throw new Error(
```

Comment:
If you throw an error, the whole page will crash. Instead, you want to gracefully degrade and still show _something_. This is returning swimlane data which is static and should not fail. What would fail is fetching the item number count but that fetch function already handles the error state. So in this case this is redundant and not what we need. 

I don't think you need this entire function at all and you can export `homepageData` and just call that in `app/page.tsx`.